testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java

```diff
@@ -123,12 +123,22 @@ public static QueryExecutor onDelta()
         // return 502 then as well. Handling this with a query retry allows us to use the cluster's autostart feature safely,
         // while keeping costs to a minimum.
 
-        RetryPolicy<QueryResult> databricksRetryPolicy = new RetryPolicy<QueryResult>()
+        RetryPolicy<QueryResult> databricks502RetryPolicy = new RetryPolicy<QueryResult>()
                 .handleIf(throwable -> throwable.getMessage().contains("HTTP Response code: 502"))
                 .withBackoff(1, 10, ChronoUnit.SECONDS)
                 .withMaxRetries(60)
                 .onRetry(event -> log.warn(event.getLastFailure(), "Query failed on attempt %d, will retry.", event.getAttemptCount()));
 
+        // Databricks JDBC driver retries operations that receive HTTP 503 responses
+        // if the server response is returned with Retry-After headers by default.
+        // Following policy retries only once when received 'HTTP retry after response' without Retry-After headers.
```

Comment:
So we want to retry "even more" than the driver does by itself, right?
I.e. we retry on top of driver's retries?

- let's make it explicit in the comment
- is there a way to configure the driver to retry for a longer time than it does by default?

Reply:
JDBC driver doesn't retry when the response doesn't have Retry-After headers. Added retry policy covers the situation. 

We can increase `TemporarilyUnavailableRetryTimeout` (900s by default) when it contains Retry-After headers. However, increasing the value won't resolve the current flaky issue.